transformer/transformer_test.go

```diff
@@ -0,0 +1,189 @@
+package transformer
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestBuildMatches(t *testing.T) {
+	Convey("Build matches for translates successfully", t, func() {
```

Comment:
It would be good to add a test for one highlight object as well as all of them; as they are optional fields remember!

I think this new test will show you the struct is wrong for `esHighlight` and that the function needs to check if esHighlights.<field> != nil` before looping through.

Reply:
This idea would apply across the `esResponse` object - the definition of mandatory and optional fields should exist in the elasticsearch mappings (set when the index is created). Use this in combination with the elasticsearch 2 documentation to determine optional fields returned by the elastic search API. We could always pair on this if needed.

Reply:
Updated functionality to expect the possibility of empty fields and I've added unit tests both individually and as a whole.